tests/problem3Tests.rb

```diff
@@ -0,0 +1,44 @@
+require 'test/unit'
+require '../src/problem3'
+
+class Problem3Tests < Test::Unit::TestCase
+
+  def setup
+    @problem = Problem3.new
+    @testDistinctValues = [1,9,2,8,3,7,4,6,5]
+    @testOrderedData = (1..100).to_a
+    @testRandomDuplicatesData = [5,9,5,8,7,4,3,7,1,4]
+    @testRandomMultipleOccurrenceDuplicatesData = [5,9,5,8,7,4,3,7,1,4,7,5]
+    @testSameValuesData = [1,1,1,1,1,1,1,1,1]
+    @testDataEmptyArray = []
+  end
+
+  def teardown
+
+  end
+
+  def test_inputDistinctValuesHasExpectedResult
```

Comment:
Your tests should be very easy to read and show the whole truth.

They basically need to broken down into 3 sections:

- prepare
- act
- assert

This sections can be even separated by a blank line.

Here is an example that I am know you will like https://msdn.microsoft.com/en-us/library/hh694602.aspx#Anchor_3